java-frontend/src/main/java/org/sonar/java/se/MethodYield.java

```diff
@@ -20,38 +20,64 @@
 package org.sonar.java.se;
 
 import org.sonar.java.se.constraint.Constraint;
-import org.sonar.java.se.symbolicvalues.SymbolicValue;
+
+import javax.annotation.Nullable;
 
 import java.util.Arrays;
-import java.util.List;
 
 public class MethodYield {
-  final List<ConstrainedSymbolicValue> parameters;
-  final ConstrainedSymbolicValue result;
+  Constraint[] parametersConstraints;
+  int resultIndex;
+  @Nullable
+  Constraint resultConstraint;
+
+  public MethodYield(MethodYield yield) {
+    this.parametersConstraints = Arrays.copyOf(yield.parametersConstraints, yield.parametersConstraints.length);
+    this.resultIndex = yield.resultIndex;
+    this.resultConstraint = yield.resultConstraint;
+  }
 
-  public MethodYield(List<ConstrainedSymbolicValue> parameters, ConstrainedSymbolicValue result) {
-    this.parameters = parameters;
-    this.result = result;
+  public MethodYield(int size) {
+    this.parametersConstraints = new Constraint[size];
+    this.resultIndex = -1;
+    this.resultConstraint = null;
   }
 
   @Override
   public String toString() {
-    return "{params: " + Arrays.toString(parameters.toArray()) + ", result: " + result + "}";
+    return "{params: " + Arrays.toString(parametersConstraints) + ", result: " + resultConstraint + "(" + resultIndex + ")}";
   }
 
-  public static class ConstrainedSymbolicValue {
-    final SymbolicValue symbolicValue;
-    final Constraint constraint;
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + Arrays.hashCode(parametersConstraints);
+    result = prime * result + ((resultConstraint == null) ? 0 : resultConstraint.hashCode());
+    result = prime * result + resultIndex;
+    return result;
+  }
 
-    public ConstrainedSymbolicValue(SymbolicValue symbolicValue, Constraint constraint) {
-      this.symbolicValue = symbolicValue;
-      this.constraint = constraint;
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 6, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1142)